### PR TITLE
Refactor usage of common interfaces in sovereign

### PR DESCRIPTION
**Pull request by SebastianMarian**
## Reasoning behind the pull request
- 
- Refactored usage of common interfaces in sovereign chain
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: testscommon/extendedShardHeaderRequestHandlerStub.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package testscommon
+
+type ExtendedShardHeaderRequestHandlerStub struct {
```
Comment (iulianpascalau) [inline testscommon/extendedShardHeaderRequestHandlerStub.go:3]: missing mock comment

File: epochStart/bootstrap/storageProcess_test.go [hunk 2]
```diff
@@ -538,3 +538,49 @@ func TestStorageEpochStartBootstrap_applyCurrentShardIDOnMiniblocksCopy(t *testi
...
+	t.Run("should create a main chain instance", func(t *testing.T) {
+		t.Parallel()
+
+		args.ChainRunType = common.ChainRunTypeRegular
+
+		sesb, _ := NewStorageEpochStartBootstrap(args)
+
+		requestHandler, err := sesb.createStorageRequestHandler()
+
+		require.NotNil(t, requestHandler)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply epochStart/bootstrap/storageProcess_test.go:557]: Done

File: factory/processing/processComponents.go [hunk 3]
```diff
@@ -717,6 +710,31 @@ func (pcf *processComponentsFactory) Create() (*processComponents, error) {
...
+		pcf.whiteListHandler,
+		common.MaxTxsToRequest,
+		pcf.bootstrapComponents.ShardCoordinator().SelfId(),
+		time.Second,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	switch pcf.chainRunType {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply factory/processing/processComponents.go:728]: :+1:

File: factory/mock/extendedShardHeaderTrackerStub.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package mock
+
+import "github.com/multiversx/mx-chain-core-go/data"
+
+type ExtendedShardHeaderTrackerStub struct {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply factory/mock/extendedShardHeaderTrackerStub.go:5]: Added

File: process/track/sovereignChainBlockProcessor.go [hunk 5]
```diff
@@ -18,14 +23,21 @@ func NewSovereignChainBlockProcessor(blockProcessor *blockProcessor) (*sovereign
...
 	}
 
 	scbp.shouldProcessReceivedHeaderFunc = scbp.shouldProcessReceivedHeader
 	scbp.processReceivedHeaderFunc = scbp.processReceivedHeader
 	scbp.doJobOnReceivedCrossNotarizedHeaderFunc = scbp.doJobOnReceivedCrossNotarizedHeader
 	scbp.requestHeaderWithShardAndNonceFunc = scbp.requestHeaderWithShardAndNonce
 
+	extendedShardHeaderRequester, ok := scbp.requestHandler.(extendedShardHeaderRequestHandler)
+	if !ok {
+		return nil, process.ErrWrongTypeAssertion
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/track/sovereignChainBlockProcessor.go:36]: Done

File: process/track/sovereignChainShardBlockTrack_test.go [hunk 6]
```diff
@@ -18,6 +18,12 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func CreateSovereignChainShardTrackerMockArguments() track.ArgShardTracker {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/track/sovereignChainShardBlockTrack_test.go:21]: Done

File: testscommon/extendedShardHeaderRequestHandlerStub.go (hunk 1 above)
Comment (SebastianMarian) [reply testscommon/extendedShardHeaderRequestHandlerStub.go:3]: Added

File: process/track/sovereignChainBlockProcessor_test.go [hunk 7]
```diff
@@ -19,6 +19,12 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func CreateSovereignChainBlockProcessorMockArguments() track.ArgBlockProcessor {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/track/sovereignChainBlockProcessor_test.go:22]: Added

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**